src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2358 @@ class GDIFileStream : public InputStream
+                m_extraBlendTileNdxes[i] = 0;
+            }
+        }
+    }
+
+    if (g_theWriteableGlobalData) {
+        if (g_theWriteableGlobalData->m_drawEntireTerrain) {
+            m_drawWidthX = m_width;
+            m_drawHeightY = m_height;
+        }
+    }
+
+    if (m_drawWidthX > m_width) {
+        m_drawWidthX = m_width;
+    }
+
+    if (m_drawHeightY > m_height) {
+        m_drawHeightY = m_height;
+    }
+
+    g_theSidesList->Validate_Sides();
+    Setup_Alpha_Tiles();
+}
+
+WorldHeightMap::~WorldHeightMap()
+{
+    if (m_data) {
+        delete[] m_data;
+        m_data = nullptr;
+    }
+
+    if (m_tileNdxes) {
+        delete[] m_tileNdxes;
+        m_tileNdxes = nullptr;
+    }
+
+    if (m_blendTileNdxes) {
+        delete[] m_blendTileNdxes;
+        m_blendTileNdxes = nullptr;
+    }
+
+    if (m_extraBlendTileNdxes) {
+        delete[] m_extraBlendTileNdxes;
+        m_extraBlendTileNdxes = nullptr;
+    }
+
+    if (m_cliffInfoNdxes) {
+        delete[] m_cliffInfoNdxes;
+        m_cliffInfoNdxes = nullptr;
+    }
+
+    if (m_cellFlipState) {
+        delete[] m_cellFlipState;
+        m_cellFlipState = nullptr;
+    }
+
+    if (m_seismicUpdateFlags) {
+        delete[] m_seismicUpdateFlags;
+        m_seismicUpdateFlags = nullptr;
+    }
+
+    if (m_siesmicZVelocity) {
+        delete[] m_siesmicZVelocity;
+        m_siesmicZVelocity = nullptr;
+    }
+
+    if (m_cliffState) {
+        delete[] m_cliffState;
+        m_cliffState = nullptr;
+    }
+
+    for (int i = 0; i < TILE_COUNT; i++) {
+        Ref_Ptr_Release(m_sourceTiles[i]);
+        Ref_Ptr_Release(m_edgeTiles[i]);
+    }
+
+    for (int i = 0; i < 12; i++) {
+        Ref_Ptr_Release(s_alphaTiles[i]);
+    }
+
+    Ref_Ptr_Release(m_terrainTex);
+    Ref_Ptr_Release(m_alphaTerrainTex);
+}
+
+void WorldHeightMap::Free_List_Of_Map_Objects()
+{
+    if (MapObject::s_theMapObjectListPtr) {
+        MapObject::s_theMapObjectListPtr->Delete_Instance();
+        MapObject::s_theMapObjectListPtr = nullptr;
+    }
+
+    MapObject::Get_World_Dict()->Clear();
 }
 
-bool WorldHeightMap::Get_Flip_State(int x, int y)
+bool WorldHeightMap::Get_Flip_State(int x, int y) const
 {
-#ifdef GAME_DLL
-    return Call_Method<bool, WorldHeightMap, int, int>(PICK_ADDRESS(0x007449F0, 0x005EBE61), this, x, y);
-#else
+    if (x < 0 || y < 0) {
+        return false;
+    }
+
+    if (y >= m_height) {
+        return false;
+    }
+
+    if (x >= m_width) {
+        return false;
+    }
+
+    if (m_cellFlipState) {
+        return ((1 << (x & 7)) & m_cellFlipState[(x >> 3) + m_flipStateWidth * y]) != 0;
+    }
+
     return false;
-#endif
+}
+
+void WorldHeightMap::Set_Flip_State(int x_index, int y_index, bool state)
+{
+    if (x_index >= 0 && y_index >= 0 && y_index < m_height && x_index < m_width) {
+        if (m_cellFlipState) {
+            unsigned char *c = &m_cellFlipState[(x_index >> 3) + m_flipStateWidth * y_index];

```

Comment:
there is no need to use a pointer here just use a reference saves a bunch of characters.
```
auto& c = m_cell....
```